doc-content/enterprise-only/cases/case-management-variable-tags-con.adoc

```diff
@@ -0,0 +1,66 @@
+[id='case-management-variable-tags-con']
+= Variable tags
+Variables store data that is used during runtime. For greater control over variable behavior, you can tag case variables and local variables in the BPMN case file. Tags are simple string values that you add as metadata to a specific variable.
+
+{PRODUCT} supports the following tags for case and local variables:
+
+* *required*: Sets the variable as a requirement in order to start a case. If a case starts without the required variable, {PRODUCT} generates a `VariableViolationException` error.
+* *readonly*: Indicates that the variable is for informational purposes only and can be set only once during case execution. If the value of a read-only variable is modified at any time, {PRODUCT} generates a `VariableViolationException` error.
+* *restricted*: A special tag that is used with the `VariableGuardProcessEventListener` to indicate that permission is granted to modify the variable based on the required and the existing role.
+
+`VariableGuardProcessEventListener` is extended from `DefaultProcessEventListener` and supports two different constructors:
+
+** `VariableGuardProcessEventListener`
++
+[source,java]
+----
+public VariableGuardProcessEventListener(String requiredRole, IdentityProvider identityProvider) {
+    this("restricted", requiredRole, identityProvider);
+}
+----
+
+** `VariableGuardProcessEventListener`
++
+[source,java]
+----
+public VariableGuardProcessEventListener(String tag, String requiredRole, IdentityProvider identityProvider) {
+    this.tag = tag;
+    this.requiredRole = requiredRole;
+    this.identityProvider = identityProvider;
+}
+----
+
++
+Therefore, you must add an event listener to the session with the allowed role name and identity provider that returns the user role as shown in the following example:
+
++
+[source,java]
+----
+ksession.addEventListener(new VariableGuardProcessEventListener("AdminRole", myIdentityProvider));
+----
+
++
+In the above example, the `VariableGuardProcessEventListener` method verifies if a variable is tagged with a security constraint tag (`restricted`). If the user does not have the required role, then {PRODUCT} generates a `VariableViolationException` error.
```

Comment:
after the required role, it can be specified (i.e, "AdminRole")

Reply:
Done :)